MobileBuy/sample/src/main/java/com/shopify/sample/activity/CollectionListActivity.java

```diff
@@ -117,8 +116,35 @@ public void failure(BuyClientError error) {
                     } else {
                         Toast.makeText(CollectionListActivity.this, "Optimizely Invalid", Toast.LENGTH_SHORT).show();
                     }
+                    }
+                });
+            } else {
+                if (initializationMode.equals("sync_datafile")) {
+                    // There is an issue with Appetize/Appium where we can't pass in the datafile through the test,
+                    // therefore we will compile it in for now and revisit once they fix that issue.
+                    // String compiledDatafile = getIntent().getStringExtra("optlyDatafile");
+                    String compiledDatafile = "{\"groups\": [], \"variables\": [], \"experiments\": [{\"status\": \"Running\", \"key\": \"product_sorting_logic\", \"layerId\": \"8089750552\", \"trafficAllocation\": [{\"entityId\": \"8095960396\", \"endOfRange\": 5000}, {\"entityId\": \"8094731902\", \"endOfRange\": 10000}], \"audienceIds\": [], \"variations\": [{\"variables\": [], \"id\": \"8095960396\", \"key\": \"sort_by_price\"}, {\"variables\": [], \"id\": \"8094731902\", \"key\": \"sort_by_name\"}], \"forcedVariations\": {\"e2e_test_user_variation_2\": \"sort_by_name\", \"e2e_test_user_variation_1\": \"sort_by_price\"}, \"id\": \"8096020192\"}, {\"status\": \"Running\", \"key\": \"product_cta\", \"layerId\": \"8224521262\", \"trafficAllocation\": [{\"entityId\": \"8221073372\", \"endOfRange\": 5000}, {\"entityId\": \"8221994870\", \"endOfRange\": 10000}], \"audienceIds\": [], \"variations\": [{\"variables\": [], \"id\": \"8221073372\", \"key\": \"get_it\"}, {\"variables\": [], \"id\": \"8221994870\", \"key\": \"buy_it\"}], \"forcedVariations\": {}, \"id\": \"8227341368\"}], \"audiences\": [], \"anonymizeIP\": true, \"accountId\": \"6384711706\", \"projectId\": \"8087040246\", \"version\": \"3\", \"attributes\": [{\"id\": \"8239264589\", \"key\": \"should_use_new_pipeline\"}, {\"id\": \"8240311689\", \"key\": \"project_size\"}], \"events\": [{\"experimentIds\": [\"8096020192\"], \"id\": \"8090781632\", \"key\": \"product_click\"}, {\"experimentIds\": [\"8096020192\"], \"id\": \"8091730243\", \"key\": \"complete_checkout\"}, {\"experimentIds\": [\"8227341368\"], \"id\": \"8227392987\", \"key\": \"start_checkout\"}], \"revision\": \"15\"}";
```

Comment:
add `TODO:` or some reminder in code or link to this in a JIRA so we have a bookmark to come back to